panoply/errors/exceptions.py

```diff
@@ -30,3 +30,8 @@ class TokenValidationException(PanoplyException):
     def __init__(self, original_error, args=None, retryable=True):
         super().__init__(args, retryable)
         self.original_error = original_error
+
+
+class MetadataNotSupportedException(Exception):
+    def __init__(self, msg: str):
+        super().__init__(msg)
```

Comment:
you better use built-in `NotImplementedError`: https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
or was it intended?